File: Java/025.java
import java.math.BigInteger;

public class Problem025 {

    public static void main(String[] args) {
        int digits = 1000;
        BigInteger f1 = BigInteger.ONE;
        BigInteger f2 = BigInteger.ONE;
        int index = 2;

        while (f2.toString().length() < digits) {
            BigInteger temp = f2;
            f2 = f2.add(f1);
            f1 = temp;
            index++;
        }

        System.out.println(index);
    }
}

File: Rust/025.rs
use num_bigint::BigUint;
use std::str::FromStr;

fn main() {
    let digits = 1000;
    let mut f1 = BigUint::from_str("1").unwrap();
    let mut f2 = BigUint::from_str("1").unwrap();
    let mut index = 2;

    while f2.to_string().len() < digits {
        let temp = f2.clone();
        f2 = &f2 + &f1;
        f1 = temp;
        index += 1;
    }

    println!("{}", index);
}
